internal/evaluator/filters.go

```diff
@@ -0,0 +1,322 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX‑License‑Identifier: Apache‑2.0
+
+package evaluator
+
+import (
+	"encoding/json"
+	"strings"
+
+	ecc "github.com/enterprise-contract/enterprise-contract-controller/api/v1alpha1"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/conforma/cli/internal/opa/rule"
+)
+
+//////////////////////////////////////////////////////////////////////////////
+// Interfaces
+//////////////////////////////////////////////////////////////////////////////
+
+// RuleFilter decides whether an entire package (namespace) should be
+// included in the evaluation set.
+//
+// The filtering system works at the package level - if any rule in a package
+// matches the filter criteria, the entire package is included for evaluation.
+// This ensures that related rules within the same package are evaluated together.
+type RuleFilter interface {
+	Include(pkg string, rules []rule.Info) bool
+}
+
+// FilterFactory builds a slice of filters for a given `ecc.Source`.
+//
+// Multiple filters can be applied simultaneously using AND logic - all filters
+// must approve a package for it to be included in the evaluation set.
+type FilterFactory interface {
+	CreateFilters(source ecc.Source) []RuleFilter
+}
+
+//////////////////////////////////////////////////////////////////////////////
+// DefaultFilterFactory
+//////////////////////////////////////////////////////////////////////////////
+
+// DefaultFilterFactory creates filters based on the source configuration.
+// It handles two main filtering mechanisms:
+// 1. Pipeline intention filtering - based on rule metadata
+// 2. Include list filtering - based on explicit package/collection names
+type DefaultFilterFactory struct{}
+
+func NewDefaultFilterFactory() FilterFactory { return &DefaultFilterFactory{} }
+
+// CreateFilters builds a list of filters based on the source configuration.
+//
+// The filtering logic follows these rules:
+// 1. Pipeline Intention Filtering:
+//   - When pipeline_intention is set in ruleData: only include packages with rules
+//     that have matching pipeline_intention metadata
+//   - When pipeline_intention is NOT set in ruleData: only include packages with rules
+//     that have NO pipeline_intention metadata (general-purpose rules)
+//
+// 2. Include List Filtering:
+//   - When includes are specified: only include packages that match the include criteria
+//   - Supports @collection, package names, and package.rule patterns
+//
+// 3. Combined Logic:
+//   - All filters are applied with AND logic - a package must pass ALL filters
+//   - This allows fine-grained control over which rules are evaluated
+func (f *DefaultFilterFactory) CreateFilters(source ecc.Source) []RuleFilter {
+	var filters []RuleFilter
+
+	// ── 1. Pipeline‑intention ───────────────────────────────────────────────
+	intentions := extractStringArrayFromRuleData(source, "pipeline_intention")
```

Comment:
Uhoh, this might be a problem.

The way we typically set the pipeline_intention is with a command line flag like this: ``ec validate image .... --extra-rule-data pipeline_intention=production`. It's not generally present in the ECP.

This code makes me think it will try to find it in the ECP directly.

Maybe we have to think of another way to detect the pipeline_indention. One quick idea (not sure if practiccal): `opa eval 'data.lib.rule_data("pipeline_intention")'` will display it.

Reply:
Another idea: We could do some creating interpretation of the `--extra-rule-data pipeline_intention=production` flag and set the value to var somewhere. That actually has some benefits in that we're moving away from the need for pipeline_intention being in the data at all.